components/screens/browser-api/sign-message/HotWalletSignMessage.tsx

```diff
@@ -0,0 +1,59 @@
+import { SignMessagePayload } from "@/api/message";
+import SignMessage from "@/components/screens/browser-api/sign-message/SignMessage";
+import { IWallet } from "@/lib/wallet/wallet-interface.ts";
+import { AccountFactory } from "@/lib/wallet/wallet-factory";
+import useWalletManager from "@/hooks/useWalletManager.ts";
+import useRpcClientStateful from "@/hooks/useRpcClientStateful";
+import Splash from "@/components/screens/Splash.tsx";
+
+type HotWalletSignMessageProps = {
+  requestId: string;
+  payload: SignMessagePayload;
+};
+
+export default function HotWalletSignMessage({
+  requestId,
+  payload,
+}: HotWalletSignMessageProps) {
+  const { getWalletSecret } = useKeyring();
+  const { wallet: walletInfo, account } = useWalletManager();
+  const { rpcClient, networkId: rpcNetworkId } = useRpcClientStateful();
+  const [wallet, setWallet] = useState<IWallet>();
+  const loading =
+    !rpcClient || !wallet || !walletInfo || !account || !rpcNetworkId;
+
+  useEffect(() => {
+    if (!rpcClient || !walletInfo || !rpcNetworkId || !account) return;
```

Comment:
Why is `account` missing here? can do `!loading`?

Reply:
I think you are mentioning `wallet`.
It is for setting `IWallet`, so unluckily, so we can't check wallet here, or the wallet would never be initialized.